threedi_modelchecker/config.py

```diff
@@ -2754,6 +2817,321 @@ def is_none_or_empty(col):
     )
 ]
 
+## 018x cross section parameters (continues 008x)
+vegetation_parameter_columns = [
+    models.CrossSectionDefinition.vegetation_drag_coefficients,
+    models.CrossSectionDefinition.vegetation_heights,
+    models.CrossSectionDefinition.vegetation_stem_diameters,
+    models.CrossSectionDefinition.vegetation_stem_densities,
+]
+CHECKS += [
+    QueryCheck(
+        error_code=180,
+        column=col,
+        invalid=Query(models.CrossSectionDefinition)
+        .filter(
+            models.CrossSectionDefinition.shape
+            != constants.CrossSectionShape.TABULATED_YZ
+        )
+        .filter(col.is_not(None)),
+        message=(
+            f"{col.table.name}.{col.name} can only be used in combination with "
+            f"a {constants.CrossSectionShape.TABULATED_YZ.name} cross section shape"
+        ),
+    )
+    for col in vegetation_parameter_columns
+    + [models.CrossSectionDefinition.friction_values]
+]
+CHECKS += [
+    CrossSectionVariableCorrectLengthCheck(
+        error_code=181,
+        column=col,
+        shapes=(constants.CrossSectionShape.TABULATED_YZ,),
+        filters=models.CrossSectionDefinition.height.is_not(None) & col.is_not(None),
+    )
+    for col in vegetation_parameter_columns
+]
+CHECKS += [
+    CrossSectionFloatListCheck(
+        error_code=187,
+        column=col,
+        shapes=(constants.CrossSectionShape.TABULATED_YZ,),
+    )
+    for col in vegetation_parameter_columns
+]
+CHECKS += [
+    QueryCheck(
+        error_code=182,
+        level=CheckLevel.WARNING,
+        column=col_csloc,
+        invalid=Query(models.CrossSectionDefinition)
+        .join(
+            models.CrossSectionLocation,
+            models.CrossSectionLocation.definition_id
+            == models.CrossSectionDefinition.id,
+        )
+        .filter(col_csloc.is_not(None) & col_csdef.is_not(None))
```

Comment:
here, again, col_csloc and col_cs_def could be renamed in line with the earlier change

Reply:
Oeps, resolved in 47b95f12d7990dd0a3c40ad54f9aaa0a1ad36819